### PR TITLE
[RFC] Run Prettier@0.21

### PR DESCRIPTION
**Pull request by hawkrives**
This is a discussion PR.

> I think I'll PR it and we can discuss it inline; then, if we like it, I can make Travis run prettier automatically for PRs.

So. Let's discuss!

> [What is Prettier?](https://github.com/prettier/prettier)
>
> Prettier is an opinionated JavaScript formatter inspired by refmt with advanced support for language features from ES2017, JSX, and Flow. It removes all original styling and ensures that all outputted JavaScript conforms to a consistent style. 
> 
> This goes way beyond eslint and other projects built on it. Unlike eslint, there aren't a million configuration options and rules. But more importantly: everything is fixable. This works because prettier never "checks" anything; it takes JavaScript as input and outputs the formatted JavaScript as output.
> 
> In technical terms: prettier parses your JavaScript into an AST and pretty-prints the AST, completely ignoring any of the original formatting. Say hello to completely consistent syntax!

### CODE REVIEW

**Comment by drewvolz** [comment]
I am all-for an automated solution that reformats to fit the `prettier` standard. I feel as if it would improve readability and help catch syntax errors. This is fine by me.

**Comment by hawkrives** [comment]
The flow error is from a bug in prettier ([issue](https://github.com/prettier/prettier/issues/877), [pr](https://github.com/prettier/prettier/pull/878)) that translates `(T|U)[]` (an array of T or U) into `T|U[]` (either T, or an array of U).

It'll be fixed in the next version, probably.

File: dangerfile.js [hunk 1]
```diff
@@ -17,7 +17,9 @@ const unFlowedFiles = jsFiles.filter(filepath => {
 })
 
 if (unFlowedFiles.length > 0) {
-  warn(`These new JS files do not have Flow enabled: ${unFlowedFiles.join(', ')}`)
+  warn(
+    `These new JS files do not have Flow enabled: ${unFlowedFiles.join(', ')}`,
```
Comment (hawkrives) [inline dangerfile.js:21]: For cases like this, you're encouraged to pull the variable bit out into a separate statement.

File: source/app.js [hunk 2]
```diff
@@ -34,14 +34,17 @@ import {FilterView} from './views/components/filter'
...
+  BuildingHoursDetailView,
+} from './views/building-hours'
 import TransportationView from './views/transportation'
 import SettingsView from './views/settings'
 import SISLoginView from './views/settings/login'
 import CreditsView from './views/settings/credits'
 import PrivacyView from './views/settings/privacy'
 import LegalView from './views/settings/legal'
-import {StudentOrgsView, StudentOrgsDetailView}  from './views/student-orgs'
+import {StudentOrgsView, StudentOrgsDetailView} from './views/student-orgs'
```
Comment (hawkrives) [inline source/app.js:47]: This is the beauty of prettier ignoring the input formatting.

File: source/app.js [hunk 3]
```diff
@@ -51,47 +54,76 @@ function renderScene(route, navigator) {
...
+    case 'EditHomeView':
+      return <EditHomeView {...props} />
+    case 'StudentOrgsView':
+      return <StudentOrgsView {...props} />
+    case 'StudentOrgsDetailView':
+      return <StudentOrgsDetailView {...props} />
+    case 'FaqView':
+      return <FaqView {...props} />
+    default:
+      return <NoRoute {...props} />
```
Comment (hawkrives) [inline source/app.js:110]: We might want to think about alternate ways of accomplishing this, I suppose.

File: source/globalize-fetch.js [hunk 4]
```diff
@@ -6,9 +6,9 @@ function status(response) {
   if (response.status >= 200 && response.status < 300) {
     return response
   } else {
-    let error = new Error(response.statusText)
+    let error = new Error(response.statusText);
     // attach the original response to the thrown error
-    ;(error: any).response = response
+    (error: any).response = response
```
Comment (hawkrives) [inline source/globalize-fetch.js:11]: I revisit this whenever we change flow versions… still haven't come up with a better solution.

File: source/lib/courses/load-all-courses.js [hunk 5]
```diff
@@ -6,9 +6,14 @@
...
 import {loadStudentNumber} from './load-student-number'
-import {loadCoursesFromUnofficialTranscript} from './load-courses-from-unofficial-transcript'
+import {
+  loadCoursesFromUnofficialTranscript,
+} from './load-courses-from-unofficial-transcript'
 
-export async function loadAllCourses(isConnected: boolean, force?: boolean): Promise<CourseCollectionType> {
+export async function loadAllCourses(
+  isConnected: boolean,
+  force?: boolean,
```
Comment (hawkrives) [inline source/lib/courses/load-all-courses.js:15]: You might ask, "Are trailing commas valid in function parameter lists?"

The answer is, "kinda". They've been accepted for ES2017 (https://github.com/tc39/proposal-trailing-function-commas), but you can't really use them yet. However, we use React Native, which compiles all our JS code anyway, and they include a compiler for the trailing function commas, so we get to use them!

All that is to say, yes, they are valid for use to use. 😄

File: source/lib/financials/balances.js [hunk 6]
```diff
@@ -6,14 +6,21 @@ import * as cache from '../cache'
 import {FINANCIALS_URL, LANDING_URL} from './urls'
 import type {FinancialDataShapeType} from './types'
 
-type PromisedDataType = Promise<{error: true, value: Error}|{error: false, value: FinancialDataShapeType}>;
-
-export async function getFinancialData(isConnected: boolean, force?: bool): PromisedDataType {
+type PromisedDataType = Promise<
+  | {error: true, value: Error}
+  | {error: false, value: FinancialDataShapeType}>;
```
Comment (hawkrives) [inline source/lib/financials/balances.js]: I'm going to file an issue about this chunk; I don't like how it breaks the contents after the `<` but leaves the closing `>` on the closing line.

File: source/views/building-hours/detail.android.js [hunk 7]
```diff
@@ -134,38 +142,66 @@ export class BuildingHoursDetailView extends React.Component {
...
                     key={i}
                     now={this.state.now}
                     schedule={schedule}
-                    isActive={set.isPhysicallyOpen !== false && schedule.days.includes(dayOfWeek) && isBuildingOpenAtMoment(schedule, this.state.now)}
-                  />)}
+                    isActive={
+                      set.isPhysicallyOpen !== false &&
+                        schedule.days.includes(dayOfWeek) &&
+                        isBuildingOpenAtMoment(schedule, this.state.now)
+                    }
```
Comment (hawkrives) [inline source/views/building-hours/detail.android.js:171]: This block is much nicer, expanded.

File: source/views/components/filter/section-list.js [hunk 8]
```diff
@@ -15,8 +15,10 @@ type PropsType = {
 
 export function ListSection({filter, onChange}: PropsType) {
   const {spec} = filter
-  const {title='', options, selected, mode} = spec
-  const {caption=`Show items with ${mode === 'AND' ? 'all' : 'any'} of these options.`} = spec
+  const {title = '', options, selected, mode} = spec
+  const {
+    caption = `Show items with ${mode === 'AND' ? 'all' : 'any'} of these options.`,
+  } = spec
```
Comment (hawkrives) [inline source/views/components/filter/section-list.js]: This … should probably be reworked into two statements.

File: source/views/components/open-url.js [hunk 9]
```diff
@@ -55,14 +58,12 @@ function iosOpen(url: string) {
...
-      enableUrlBarHiding: true,
-      enableDefaultShare: true,
-    })
-    // fall back to opening in Chrome / Browser / platform default
-    .catch(() => genericOpen(url))
+  return CustomTabs.openURL(url, {
+    showPageTitle: true,
+    enableUrlBarHiding: true,
+    enableDefaultShare: true,
+  })// fall back to opening in Chrome / Browser / platform default
```
Comment (hawkrives) [inline source/views/components/open-url.js]: Comments are still a little tricky sometimes, but they keep getting better.

File: source/views/components/tabbed-view/index.ios.js [hunk 10]
```diff
@@ -25,36 +25,41 @@ export default class TabbedView extends React.Component {
...
-            onPress={() => this.onChangeTab(tab.id)}
-          >
-            <tab.component
-              {...this.props.childProps}
-              {...(tab.props || {})}
-              {...baseProps}
-            />
-          </Icon.TabBarItemIOS>
+          return (
+            <Icon.TabBarItemIOS
```
Comment (hawkrives) [inline source/views/components/tabbed-view/index.ios.js:46]: THE CONSISTENCY 😭 IT IS SO PRETTY

File: source/views/home/button.js [hunk 11]
```diff
@@ -28,7 +29,7 @@ export function HomeScreenButton({view, onPress}: {view: ViewType, onPress: () =
 export const CELL_MARGIN = 10
 const cellVerticalPadding = 8
 const cellHorizontalPadding = 4
-const cellWidth = (Viewport.width / 2) - (CELL_MARGIN * 1.5)
+const cellWidth = Viewport.width / 2 - CELL_MARGIN * 1.5
```
Comment (hawkrives) [inline source/views/home/button.js:32]: It's gonna make you learn those arithmetical orders of operations, though 😝

File: source/views/menus/lib/__tests__/menu-item.mock.js [hunk 12]
```diff
@@ -1,23 +1,27 @@
...
   options: [],
   rating: '',
   special: special,
   station: station,
-  'sub_station': '',
-  'sub_station_order': '',
-  'sub_station_id': '',
+  sub_station: '',
+  sub_station_order: '',
+  sub_station_id: '',
```
Comment (hawkrives) [inline source/views/menus/lib/__tests__/menu-item.mock.js:24]: These… I guess I'll add eslint-ignore comments to these spots to ignore the camelcase warnings.

File: source/views/news/fetch-feed.js [hunk 13]
```diff
@@ -15,14 +15,24 @@ import type {
...
 }
 
-export async function fetchWpJson(url: string, query?: Object): Promise<StoryType[]> {
-  const feed: WpJsonResponseType = await fetchJson(`${url}?${qs.stringify(query)}`)
+export async function fetchWpJson(
+  url: string,
+  query?: Object,
+): Promise<StoryType[]> {
+  const feed: WpJsonResponseType = await fetchJson(
+    `${url}?${qs.stringify(query)}`,
```
Comment (hawkrives) [inline source/views/news/fetch-feed.js:34]: This block got a bit awkward.

File: source/views/settings/sections/login-credentials.js [hunk 14]
```diff
@@ -44,41 +42,41 @@ class CredentialsLoginSection extends React.Component {
...
   render() {
     let {loggedIn, message} = this.props
     let {loading, username, password} = this.state
 
     return (
       <Section
-        header='ST. OLAF LOGIN'
-        footer='St. Olaf login enables the "meals remaining" feature.'
+        header="ST. OLAF LOGIN"
+        footer="St. Olaf login enables the &quot;meals remaining&quot; feature."
```
Comment (hawkrives) [inline source/views/settings/sections/login-credentials.js:58]: Turns out that JSX uses HTML's entity parsing rules

File: source/views/sis/balances.js [hunk 15]
```diff
@@ -41,16 +38,16 @@ const buttonStyles = StyleSheet.create({
...
   props: TopLevelViewPropsType & {
     flex: ?number,
     ole: ?number,
     print: ?number,
     weeklyMeals: ?number,
     dailyMeals: ?number,
-    tokenValid: bool,
-    credentialsValid: bool,
+    tokenValid: boolean,
+    credentialsValid: boolean,
```
Comment (hawkrives) [inline source/views/sis/balances.js:50]: CONSISTENCY CONSISTENCY CONSISTENCY

I feel like Steve Ballmer yelling about developers.

File: source/views/transportation/bus/lib/__tests__/get-schedule-for-now.test.js [hunk 16]
```diff
@@ -7,32 +7,142 @@ function buildBusSchedules(): BusScheduleType[] {
   return [
     {
       days: ['Mo', 'Tu'],
-      stops: ['St. Olaf', 'Carleton', 'Food Co-op', 'Cub/Target', 'El Tequila', 'Food Co-op', 'Carleton', 'St. Olaf'],
-      times: [['4:15pm',  '4:22pm',   '4:23pm',     '4:33pm',     '4:37pm',     '4:43pm',     '4:44pm',   '4:52pm'],
-              ['4:55pm',  '5:02pm',   '5:03pm',     '5:13pm',     '5:17pm',     '5:23pm',     '5:24pm',   '5:32pm'],
-              ['5:35pm',  '5:42pm',   '5:43pm',     '5:53pm',     '5:57pm',     '6:03pm',     '6:04pm',   '6:12pm'],
```
Comment (hawkrives) [inline source/views/transportation/bus/lib/__tests__/get-schedule-for-now.test.js]: This chunk should get a prettier-ignore comment.

File: source/views/transportation/bus/lib/__tests__/get-set-of-stops-for-now.test.js [hunk 17]
```diff
@@ -5,10 +5,47 @@ import type {BusScheduleType} from '../../types'
 function makeSchedule(): BusScheduleType {
   return {
     days: ['Mo', 'Tu'],
-    stops: ['St. Olaf', 'Carleton', 'Food Co-op', 'Cub/Target', 'El Tequila', 'Food Co-op', 'Carleton', 'St. Olaf'],
-    times: [['4:15pm',  '4:22pm',   '4:23pm',     '4:33pm',     '4:37pm',     '4:43pm',     '4:44pm',   '4:52pm'],
-            ['4:55pm',  '5:02pm',   '5:03pm',     '5:13pm',     '5:17pm',     '5:23pm',     '5:24pm',   '5:32pm'],
-            ['5:35pm',  '5:42pm',   '5:43pm',     '5:53pm',     '5:57pm',     '6:03pm',     '6:04pm',   '6:12pm'],
```
Comment (hawkrives) [inline source/views/transportation/bus/lib/__tests__/get-set-of-stops-for-now.test.js]: Also needs a prettier-ignore

File: source/lib/financials/balances.js (hunk 6 above)
Comment (hawkrives) [reply source/lib/financials/balances.js]: I've decided to just split this into two statements:

```js
type DataType = 
  | {error: true, value: Error} 
  | {error: false, value: Value};

type PromisedDataType = Promise<DataType>;
```

**Review by drewvolz — APPROVED** [review]

**Pull request closed without merging**